python/add_corona.py

```diff
@@ -0,0 +1,173 @@
+#!/usr/bin/env python3
+
+#===================================================
+#
+#    Copyright (c) 2025
+#      SMASH Hybrid Team
+#
+#    GNU General Public License (GPLv3 or later)
+#
+#===================================================
+
+import argparse
+import numpy as np
+import sys
+import os
+import subprocess
+
+'''
+    This script adds the corona particles from the initial SMASH run
+    and possibly others to the sampled particle list.
+'''
+
+def find_out_event_number(filepath):
+    '''
+        Look into the last line of the file and find the number of events
+    '''
+    if not os.path.isfile(filepath):
+        print(filepath+" does not exist!", file=sys.stderr)
+        sys.exit(2)
+    tail = subprocess.run(['tail', '-n', '1', filepath],
+                          stdout=subprocess.PIPE,stderr=subprocess.PIPE)
+    last_line = tail.stdout.decode().strip().split()
+    n_events = 0
+    error = False
+    if "end" in last_line:
+        try:
+            n_events = int(last_line[2])
+        except ValueError:
+            error = True
+    else:
+        error = True
+    if error:
+        print("The number of events cannot be found for " + filepath,
+                file=sys.stderr)
+        sys.exit(3)
+    return n_events+1
+
+def extract_particles(filename):
+    '''
+        Extract particles from a file to a numpy array
+    '''
+    particles = np.empty(shape=[0, 13])
+    n_event = 0
+
+    for line in open(filename, "r"):
```

Comment:
If this opening method is chosen, the file should be closed manually if I'm not mistaken. You could use
```suggestion
    with open(filename, "r") as f:
        for line in f:
```
instead, which closes the file automatically.